### PR TITLE
Json encoding for ReferencableList and testing catch-up

### PR DESCRIPTION
**Pull request by piotrswigon**
I wanted to create Json encoding for ReferencableList, that's was one line... but then I wanted to write a test, and to do properly I had to do some more stuff:

* Separated monolithic `DevToolsMessageTest` into tests for respective files.
* Dropped `StoreMessage` class, as it didn't contain anything apart from extension functions on various storage related classes, I've moved these `toJson()` extension functions to `JsonEncodings` file.
* Written tests for everything in the `JsonEncodings`, including the ReferencableList case.

And I actually found a bug (creationTimestamp was encoded as an expirationTimestamp), and one omission: we didn't encode the id of the RawEntity.

Also renamed `VERSIONMAP` to `VERSION_MAP` to maintain mapping between casing (as the original is `versionMap`).

cc @SHeimlich

### CODE REVIEW

File: java/arcs/android/devtools/JsonEncodings.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
+package arcs.android.devtools
```
Comment (jasonwyatt) [inline java/arcs/android/devtools/JsonEncodings.kt:1]: We'll need the copyright comment here...

File: java/arcs/android/devtools/JsonEncodings.kt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+/**
+ * Turn the [VersionMap] into a [JsonObject].
+ */
+fun VersionMap.toJson() = JsonObject(
+  actors.map { it to JsonNumber(this[it].toDouble()) }.toMap()
+)
+
+/**
+ * Transform a [Referencable] to Json.
```
Comment (jasonwyatt) [inline java/arcs/android/devtools/JsonEncodings.kt:26]:
```suggestion
 * Transform a [Referencable] into a [JsonValue].
```

File: javatests/arcs/android/devtools/JsonEncodingsTest.kt [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    assertThat(
+      RawEntity(
+        id = "a:b:c",
+        singletons = mapOf(
+          "nickname" to "Ollie".toReferencable(),
+          "isCool" to true.toReferencable()
+        ),
+        collections = mapOf(
+          "names" to setOf("Brian".toReferencable(), "John".toReferencable())
+        ),
```
Comment (jasonwyatt) [inline javatests/arcs/android/devtools/JsonEncodingsTest.kt:63]: Is it worth testing that singletons/collections within an Entity with values of type `Reference` can be Jsonified as well as primitives?  Also maybe ordered-lists?

File: javatests/arcs/android/devtools/StoreOperationMessageTest.kt [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4::class)
+class StoreOperationMessageTest {
+
+  @Test
+  fun toJson() = runBlockingTest {
```
Comment (jasonwyatt) [inline javatests/arcs/android/devtools/StoreOperationMessageTest.kt:32]: Not sure `runBlockingTest` is needed here? Are any of the operations in this test suspending?

**Review by jasonwyatt — APPROVED** [review]

File: javatests/arcs/android/devtools/StoreOperationMessageTest.kt (hunk 4 above)
Comment (piotrswigon) [reply javatests/arcs/android/devtools/StoreOperationMessageTest.kt]: Removed.

File: javatests/arcs/android/devtools/JsonEncodingsTest.kt (hunk 3 above)
Comment (piotrswigon) [reply javatests/arcs/android/devtools/JsonEncodingsTest.kt]: Added.

File: java/arcs/android/devtools/JsonEncodings.kt (hunk 2 above)
Comment (piotrswigon) [reply java/arcs/android/devtools/JsonEncodings.kt]: Fixed.

File: java/arcs/android/devtools/JsonEncodings.kt (hunk 1 above)
Comment (piotrswigon) [reply java/arcs/android/devtools/JsonEncodings.kt:12]: Added here and elsewhere.

**Pull request closed without merging**